src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -90,72 +90,90 @@ class oopDesc;
 // doesn't provide appropriate definitions, they should be provided in
 // the compiler-specific definitions file (e.g., globalDefinitions_gcc.hpp)
 
-#define BOOL_TO_STR(_b_) ((_b_) ? "true" : "false")
+// Guide to the suffixes used in the format specifiers for integers:
+//        - print the decimal value:                   745565
+//  _X    - print as hexadecimal, without leading 0s: 0x12345
+//  _X_0  - print as hexadecimal, with leading 0s: 0x00012345
+//  _H    - print as hexadecimal, without 0x prefix
+//  _W(w) - prints w sized string with the given value right
+//          adjusted. Use -w to print left adjusted.
```

Comment:
pre-existing: The "width" values for "W" suffixed formatters probably ought to be in the expansion as "XSTR(width)" rather than "#width", to permit the use of a macro for the width value.